src/SnowflakeTransformation.php

```diff
@@ -34,37 +38,24 @@ public function __construct(Config $config, LoggerInterface $logger)
         $this->connection = new Connection($this->databaseConfig);
     }
 
+    /**
+     * @param array<array{source: string}> $tableNames
+     * @param ManifestManager $manifestManager
+     */
     public function createManifestMetadata(array $tableNames, ManifestManager $manifestManager): void
     {
         $tableStructures = $this->getTables($tableNames);
-        foreach ($tableStructures as $tableStructure) {
+        foreach ($tableStructures as $tableDef) {
             $columnsMetadata = (object) [];
-            $columnNames = [];
-            foreach ($tableStructure['columns'] as $column) {
-                $columnNames[] = $column['name'];
-                $datatypeKeys = array_flip(['length', 'nullable']);
-                try {
-                    $datatype = new SnowflakeDatatype(
-                        $column['type'],
-                        array_intersect_key($column, $datatypeKeys)
-                    );
-                } catch (InvalidTypeException $e) {
-                    unset($column['length']);
-                    $datatype = new GenericDatatype(
-                        $column['type'],
-                        array_intersect_key($column, $datatypeKeys)
-                    );
-                }
-                $columnsMetadata->{$column['name']} = $datatype->toMetadata();
+            /** @var SnowflakeColumn $column */
+            foreach ($tableDef->getColumnsDefinitions() as $column) {
+                $columnsMetadata->{$column->getColumnName()} = $column->getColumnDefinition()->toMetadata();
             }
-            unset($tableStructure['columns']);
             $tableMetadata = [];
-            foreach ($tableStructure as $key => $value) {
-                $tableMetadata[] = [
-                    'key' => 'KBC.' . $key,
-                    'value' => $value,
-                ];
-            }
```

Comment:
To sem nepochopil řekl bych, že tam nic jiného jak name není.

Reply:
tohle ještě v pondělí vyzkouším

Reply:
jo cajk 👍